tests/components/emoncms/test_config_flow.py

```diff
@@ -0,0 +1,117 @@
+"""Test emoncms config flow."""
+
+import copy
+from typing import Any
+from unittest.mock import AsyncMock
+
+from homeassistant.components.emoncms.const import CONF_ONLY_INCLUDE_FEEDID, DOMAIN
+from homeassistant.config_entries import SOURCE_IMPORT, ConfigFlowResult
+from homeassistant.const import CONF_API_KEY, CONF_ID, CONF_PLATFORM, CONF_URL
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResultType
+
+from . import setup_integration
+from .conftest import EMONCMS_FAILURE, FLOW_RESULT, SENSOR_NAME
+
+from tests.common import MockConfigEntry
+
+YAML = {
+    CONF_PLATFORM: "emoncms",
+    CONF_API_KEY: "my_api_key",
+    CONF_ID: 1,
+    CONF_URL: "http://1.1.1.1",
+}
+
+
+async def flow_import(hass: HomeAssistant, yaml: dict[str, Any]) -> ConfigFlowResult:
+    """Import of a yaml config."""
+    return await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": SOURCE_IMPORT},
+        data=yaml,
+    )
```

Comment:
I think it makes sense to inline this tbh

Reply:
removed and reintegrated in the tests